lib/camp3/request.rb

```diff
@@ -32,44 +43,69 @@ def self.parse(body)
 
     # Decodes a JSON response into Ruby object.
     def self.decode(response)
-      response ? JSON.parse(response) : {}
+      response ? JSON.load(response) : {}
     rescue JSON::ParserError
       raise Error::Parsing, 'The response is not a valid JSON'
     end
 
     %w[get post put delete].each do |method|
       define_method method do |path, options = {}|
-        override_path = options.delete(:override_path)
         params = options.dup
-
+        override_path = params.delete(:override_path)
```

Comment:
This needs to come after the params duplication, otherwise retries with a new access token, won't have access to the `override_path`